utils/citizen_id.ts

```diff
@@ -86,9 +90,29 @@ export const withCitizenIdCheck = (
                 })
               )
         )
-    : verifySupportToken(publicRsaCertificate, citizenId).map(_ =>
-        FiscalCodeAndCitizenIdType.encode({
-          citizenIdType: "SupportToken",
-          fiscalCode: _
-        })
-      );
+    : verifySupportToken(publicRsaCertificate, citizenId)
+        // Check if the token is backlisted
+        .chain(_ =>
+          taskify<Error, number>(cb =>
+            redisClient.exists(
+              `${BLACKLIST_SUPPORT_TOKEN_PREFIX}${citizenId}`,
+              cb
+            )
+          )()
+            .chain(
+              fromPredicate(
+                existsResponse => existsResponse === 0,
+                () => new Error("Blacklisted support_token")
+              )
+            )
+            .map(_1 => _)
+            .mapLeft<
```

Comment:
You can use `bimap` instead of double mapping :)

Reply:
Fixed in b18070a